pkg/chunked/storage_linux.go

```diff
@@ -1729,9 +1752,16 @@ func (c *chunkedDiffer) ApplyDiff(dest string, options *archive.TarOptions, diff
 
 	var missingParts []missingPart
 
-	output.UIDs, output.GIDs = collectIDs(toc.Entries)
+	var entries []fileMetadata
+	for _, e := range toc.Entries {
```

Comment:
(This works just fine; alternatively, maybe it could be the job of `mergeTocEntries` to convert from the one type to the other.

That way, the semantics could be separated by type: `[]internal.FileMetadata` can contain top-level `TypeChunk` entries, and `[]fileMetadata` never contains top-level `TypeChunk`, they are all collected in `.Chunks`.)